### PR TITLE
[Docs] Link to Elasticsearch labs on landing pages

### PR DESCRIPTION
**Pull request by leemthompo**
First PR in this repo and first line of `reStructuredText` written 😄.

Adds links to Elasticsearch labs in two places:

- readthedocs landing page
- Elastic docs landing page

ℹ️ Not sure about how labels work for this repo!

### CODE REVIEW

**Review by szabosteve — APPROVED** [review]
Awesome, @leemthompo! Very nice addition! LGTM!

File: docs/sphinx/index.rst [hunk 1]
```diff
@@ -65,6 +65,12 @@ Example Usage
         print("%(timestamp)s %(author)s: %(text)s" % hit["_source"])
 
 
+Interative examples
```
Comment (pquentin) [inline docs/sphinx/index.rst:68]:
```suggestion
Interactive examples
```

**Review by pquentin — COMMENTED** [review]
Great addition! Would it help to mention that the focus is on, well, AI?

File: docs/sphinx/index.rst [hunk 2]
```diff
@@ -65,6 +65,12 @@ Example Usage
         print("%(timestamp)s %(author)s: %(text)s" % hit["_source"])
 
 
+Interactive examples
+-------------------
+
+The `elasticsearch-labs <https://github.com/elastic/elasticsearch-labs>`_ repo contains interactive and executable Python notebooks, sample apps, and resources for testing out Elasticsearch, using the Python client.
```
Comment (leemthompo) [inline docs/sphinx/index.rst:71]:
```suggestion
The `elasticsearch-labs <https://github.com/elastic/elasticsearch-labs>`_ repo contains interactive and executable Python notebooks, sample apps, and resources for testing out Elasticsearch, using the Python client. These examples are mainly focused on vector search, hybrid search and generative AI use cases, but you'll also find examples of basic operations like creating index mappings and performing lexical search.
```

Comment (leemthompo) [reply docs/sphinx/index.rst:71]: @pquentin @szabosteve how about this iteration?

Comment (pquentin) [reply docs/sphinx/index.rst:71]: This sounds great! :+1:

File: docs/sphinx/index.rst [hunk 3]
```diff
@@ -65,6 +65,12 @@ Example Usage
         print("%(timestamp)s %(author)s: %(text)s" % hit["_source"])
 
 
+Interactive examples
+-------------------
```
Comment (pquentin) [inline docs/sphinx/index.rst:69]: Oops, I missed this:

```suggestion
--------------------
```

Comment (leemthompo) [reply docs/sphinx/index.rst]: Thanks! ::Shakes fist at reStructuredText::

**Review by pquentin — DISMISSED** [review]
Thanks for iterating! LGTM.

**Review by szabosteve — APPROVED** [review]
LGTM2! 🎉 
It would be nice to have @JoshMock opinion, too, as the Py client is also his area.

**Review by JoshMock — APPROVED** [review]

**Comment by JoshMock** [comment]
This all looks great to me. :+1:

**Pull request merged**